kratos/solving_strategies/schemes/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,805 @@
+                        p_variable_data->Name());
+                rComm.AssembleCurrentData(r_variable);
+            }
+            else if (KratosComponents<Variable<double>>::Has(p_variable_data->Name()))
+            {
+                const auto& r_variable =
+                    KratosComponents<Variable<double>>::Get(p_variable_data->Name());
+                rComm.AssembleCurrentData(r_variable);
+            }
+            else
+            {
+                KRATOS_ERROR << "Variable \"" << p_variable_data->Name() << "\" not found!\n";
+            }
+        }
+        KRATOS_CATCH("");
+    }
+
+    void UpdateAcceleration(ModelPart& rModelPart)
+    {
+        if (!mUpdateAcceleration)
+            return;
+
+        ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+        const int number_of_elements = rModelPart.NumberOfElements();
+
+        auto acceleration_vars = GatherVariables(
+            rModelPart.Elements(),
+            [](const DerivativesExtension& rExtensions,
+               std::vector<const VariableData*>& rVec, ProcessInfo& rCurrentProcessInfo) {
+                rExtensions.GetSecondDerivativesVariables(rVec, rCurrentProcessInfo);
+            },
+            r_current_process_info);
+#pragma omp parallel for
+        for (int i = 0; i < number_of_elements; ++i)
+        {
+            Element& r_element = *(rModelPart.ElementsBegin() + i);
+            Geometry<Node<3>>& r_geometry = r_element.GetGeometry();
+            DerivativesExtension& r_extensions = *r_element.GetValue(DERIVATIVES_EXTENSION);
+
+            const int k = OpenMPUtils::ThisThread();
+
+            for (unsigned int i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+            {
+                r_extensions.GetFirstDerivativesVector(
+                    i_node, mIndirectCurrentVelocityVector[k], 0, r_current_process_info);
+                r_extensions.GetFirstDerivativesVector(
+                    i_node, mIndirectOldVelocityVector[k], 1, r_current_process_info);
+                r_extensions.GetSecondDerivativesVector(
+                    i_node, mIndirectCurrentAccelerationVector[k], 0, r_current_process_info);
+                r_extensions.GetSecondDerivativesVector(
+                    i_node, mIndirectOldAccelerationVector[k], 1, r_current_process_info);
+
+                r_geometry[i_node].SetLock();
+                for (unsigned int i_var = 0;
+                     i_var < mIndirectCurrentAccelerationVector[k].size(); ++i_var)
+                    mIndirectCurrentAccelerationVector[k][i_var] =
+                        mBossak.C2 * (mIndirectCurrentVelocityVector[k][i_var] -
+                                      mIndirectOldVelocityVector[k][i_var]) -
+                        mBossak.C3 * mIndirectOldAccelerationVector[k][i_var];
+                r_geometry[i_node].UnSetLock();
+            }
+        }
+        AssembleVariables(acceleration_vars, rModelPart.GetCommunicator());
```

Comment:
Either you need to synchronize here or do nothing in the case that the variables are already synchronized. This needs to be tested with MPI since serial tests won't show anything and it is a real pain to debug later on.

Reply:
updated as discussed